maze_transformer/evaluation/path_evals.py

```diff
@@ -41,29 +45,20 @@ class PathEvals:
     @register_method(evals)
     @staticmethod
     def node_overlap(solution: MazePath, prediction: MazePath, **_) -> float:
-        """number of shared nodes (any order) / total number of (unique) nodes"""
-        if len(prediction) <= 1:
-            return 0.0
+        """number of shared nodes (any order) / total number of (unique) nodes in solution"""
 
-        n_shared: int = 0
         solution_set = {tuple(coord) for coord in solution}
         prediction_set = {tuple(coord) for coord in prediction}
 
-        for coord in solution:
-            if tuple(coord) in prediction_set:
-                n_shared += 1
-        return n_shared / len(solution_set)
+        return len(prediction_set & solution_set) / len(solution_set)
 
     @register_method(evals)
     @staticmethod
     def num_connections_adjacent_lattice(prediction: MazePath, **_) -> float:
         """number of the connections in prediction which actually connect nodes that are adjacent on the lattice, ignoring if they are adjacent on the maze"""
-        if len(prediction) <= 1:
-            return 0.0
-
-        n_adj: int = 0
+        n_adj = 0.0
```

Comment:
no type hints :'(

This is definitely a nitpick but I think it helps us with making mypy checking be useful down the line